### PR TITLE
Don't sync automation logs from prod to dev app

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
Adding a test case for the removal of automation logs from app sync, as well as adding to the publish/sync filter a check to not carry over automation logs.

This was causing memory issues in very large apps containing say 300K+ automation logs and there isn't much (if any) benefit to syncing.

Addresses: 
- https://linear.app/budibase/issue/BUDI-7188/remove-syncing-of-automation-history-from-production-to-development

### CODE REVIEW

File: packages/server/src/api/routes/tests/application.spec.ts [hunk 1]
```diff
@@ -324,4 +324,39 @@ describe("/applications", () => {
       expect(events.app.unpublished).toBeCalledTimes(1)
     })
   })
+
+  describe("app sync", () => {
```
Comment (Rory-Powell) [inline packages/server/src/api/routes/tests/application.spec.ts:328]: Preference: `describe` blocks cover the endpoint they are testing e.g. 

```ts
describe("POST /api/applications/:appId/sync", () => {})
```

File: packages/server/src/api/routes/tests/application.spec.ts [hunk 2]
```diff
@@ -324,4 +324,39 @@ describe("/applications", () => {
...
+        .expect(200)
+
+      // does exist in prod
+      await context.doInAppContext(config.getProdAppId(), async () => {
+        const logs = await config.getAutomationLogs()
+        expect(logs.data.length).toBe(1)
+      })
+
+      // delete prod app so we revert to dev log search
+      await config.unpublish()
```
Comment (Rory-Powell) [inline packages/server/src/api/routes/tests/application.spec.ts:353]: This seems like pretty un-deterministic behaviour, is there not a way retrieve the desired logs by id?

File: packages/server/src/tests/utilities/structures.ts [hunk 3]
```diff
@@ -241,6 +243,21 @@ export function collectAutomation(tableId?: string): Automation {
   return automation as Automation
 }
 
+export function basicAutomationLog(automationId: string): AutomationResults {
```
Comment (Rory-Powell) [inline packages/server/src/tests/utilities/structures.ts:246]: Naming here is somewhat confusing. Is it an `AutomationLog` or `AutomationResults`?

File: packages/server/src/api/routes/tests/application.spec.ts (hunk 2 above)
Comment (mike12345567) [reply packages/server/src/api/routes/tests/application.spec.ts:353]: No - the issue isn't by ID, its by database, the way the system is designed it retrieves from prod then from dev. I can probably make this more clear in the test as the context shifts aren't really needed.

File: packages/server/src/tests/utilities/structures.ts (hunk 3 above)
Comment (mike12345567) [reply packages/server/src/tests/utilities/structures.ts:246]: Good point, yes this returns an automation result, mis-typed the function name!

File: packages/server/src/api/routes/tests/application.spec.ts (hunk 2 above)
Comment (Rory-Powell) [reply packages/server/src/api/routes/tests/application.spec.ts:353]: By "id" I mean the app id vs the prod id - essentially the different databases. 

I think the logic is just unclear? Needing to remove the prod db in order to query the dev db logs seems counter intuitive. Often tests highlight things like this which can be an opportunity to improve the interface. 

NAB = just wondered if there was an easier way to get the dev db logs.

Comment (mike12345567) [reply packages/server/src/api/routes/tests/application.spec.ts:351]: Yeah - there isn't a particularly nice way to do this - without basically writing a bunch of code into the SDK to provide access to something that isn't supposed to be accessible.

**Pull request merged**